moto/awslambda/models.py

```diff
@@ -692,6 +694,18 @@ def create_event_source_mapping(self, spec):
                     queue.lambda_event_source_mappings[esm.function_arn] = esm
 
                     return esm
+        try:
+            stream = json.loads(dynamodbstreams_backends[self.region_name].describe_stream(spec['EventSourceArn']))
+            spec.update({'FunctionArn': func.function_arn})
+            esm = EventSourceMapping(spec)
+            self._event_source_mappings[esm.uuid] = esm
+            table_name = stream['StreamDescription']['TableName']
+            table = dynamodb_backends2[self.region_name].get_table(table_name)
+            table.lambda_event_source_mappings[esm.function_arn] = esm
+
+            return esm
+        except Exception:
```

Comment:
Can we make this more specific to the exceptions that can be thrown here?

Reply:
Rewritten this so that it doesn't throw an exception at all